### PR TITLE
xds: use the correct resource id template as per xDS server gRFC

### PR DESCRIPTION
**Pull request by sanjaypujare**
To match the spec in https://github.com/grpc/proposal/blob/master/A36-xds-for-servers.md#xds-protocol .

### CODE REVIEW

File: xds/src/test/java/io/grpc/xds/XdsClientWrapperForServerSdsTestMisc.java [hunk 1]
```diff
@@ -202,6 +202,17 @@ public void startXdsClient_expectException() {
     verify(mockServerWatcher, never()).onError(any(Throwable.class), eq(false));
   }
 
+  @Test
+  public void xdsClientStart_multipleReplacements() {
```
Comment (voidzcy) [inline xds/src/test/java/io/grpc/xds/XdsClientWrapperForServerSdsTestMisc.java:206]: I  don't see any value of such a test, it's majorly testing the functionality of `String.replaceAll(...)`...

File: xds/src/main/java/io/grpc/xds/BootstrapperImpl.java [hunk 2]
```diff
@@ -215,7 +215,8 @@ BootstrapInfo bootstrap(Map<String, ?> rawData) throws XdsInitializationExceptio
         certProviders.put(name, certificateProviderInfo);
       }
     }
-    String grpcServerResourceId = JsonUtil.getString(rawData, "grpc_server_resource_name_id");
+    String grpcServerResourceId =
```
Comment (voidzcy) [inline xds/src/main/java/io/grpc/xds/BootstrapperImpl.java:218]: You should also change the property name in `BoootstrapInfo` and everywhere else.

File: xds/src/test/java/io/grpc/xds/XdsClientWrapperForServerSdsTestMisc.java (hunk 1 above)
Comment (sanjaypujare) [reply xds/src/test/java/io/grpc/xds/XdsClientWrapperForServerSdsTestMisc.java:206]: The test verifies the contract and is useful and is not there to test functionality of replaceAll. I want the test to show breakage when implementation changes.

File: xds/src/main/java/io/grpc/xds/BootstrapperImpl.java (hunk 2 above)
Comment (sanjaypujare) [reply xds/src/main/java/io/grpc/xds/BootstrapperImpl.java:218]: The names are really internal names between the Bootstrapper and the consumer (XdsClientWrapperForServerXds) so not strictly an API exposed to outside. But I will change it

Comment (sanjaypujare) [reply xds/src/main/java/io/grpc/xds/BootstrapperImpl.java:218]: BTW, what name will work for you? Is `serverListenerResourceNameTemplate` okay?

**Review by voidzcy — APPROVED** [review]

**Pull request merged**